File: ios/NetworkKit/NetworkKit/Sources/Model/Network/NetworkError.swift
import ExtensionKit

public enum NetworkError: String, ErrorStringConvertible, Codable {
    case invalidData
    case invalidResponse
    case invalidRequest

    public var description: String {
        switch self {
        case .invalidRequest:
            return "Invalid request, please try again later."
        case .invalidResponse:
            return "Invalid response, please try again later."
        case .invalidData:
            return "Invalid data, please try again later."
        }
    }
}

File: android/SwiftKotlination/modules/network/src/main/kotlin/fr/jhandguy/network/model/network/NetworkError.kt
package fr.jhandguy.network.model.network

sealed class NetworkError : Error() {
    class InvalidRequest : NetworkError()
    class InvalidResponse : NetworkError()

    override val message: String
        get() = when (this) {
            is InvalidRequest -> "Invalid request, please try again later."
            is InvalidResponse -> "Invalid response, please try again later."
        }
}
